src/yunohost/hook.py

```diff
@@ -311,14 +313,56 @@ def hook_exec(path, args=None, raise_on_error=False, no_trace=False,
         user -- User with which to run the command
 
     """
-    from moulinette.utils.process import call_async_output
 
     # Validate hook path
     if path[0] != '/':
         path = os.path.realpath(path)
     if not os.path.isfile(path):
         raise YunohostError('file_does_not_exist', path=path)
 
+    # Check the type of the hook (bash by default)
+    hook_type = "bash"
+    # (non-bash hooks shall start with something like "#!/usr/bin/env language")
+    hook_ext = os.path.splitext(path)[1]
+    if hook_ext == ".py":
+        hook_type = "python"
+    else:
+        # TODO / FIXME : if needed in the future, implement support for other
+        # languages...
+        assert hook_ext == "", "hook_exec only supports bash and python hooks for now"
```

Comment:
Maybe we could add the support for the `.sh` extension if we want to specify that it's a bash hook.